pkg/types/azure/validation/platform.go

```diff
@@ -40,6 +40,9 @@ var (
 
 	// tagKeyPrefixRegex is for verifying that the tag value does not contain restricted prefixes.
 	tagKeyPrefixRegex = regexp.MustCompile(`^(?i)(name$|kubernetes\.io|openshift\.io|microsoft|azure|windows)`)
+
+	// subscriptionIDRegex is for verifying the identity key associated with the managed key provided.
+	subscriptionIDRegex = regexp.MustCompile(`^([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$`)
```

Comment:
Did you base this on some Azure docs? It would be good to include a link to it.
Ideally, I think we should validate we can access that subscription in the installconfig validation instead of relying on a regex.

Reply:
I reckon this checks for the standard UUID regex. @jhixson74 added significantly to the code here, any thoughts?

Reply:
This is  close but should also account for upper case. You can use this one: https://github.com/openshift/installer/blob/7956adbe7f21c37f57bbc3ba3401d7b595eddc81/pkg/types/azure/validation/disk.go#L16

Reply:
Fixed it